### ISSUE TITLE
Enemies Never Drop Max Amount of Item

### ISSUE THREAD

**Issue by AntumDeluge** [arianne/stendhal#192]
Discovered a bug where enemies never drop max amount of an item. For example [madaram healer](https://stendhalgame.org/creature/madaram_healer.html) is supposed to drop up to 43 money. However, the most it ever drops is 42.

I tested this further by changing the line:
```xml
<item value="money" quantity="[5,43]" probability="100.0"/>
```
in ***data/conf/creatures/human.xml*** to:
```xml
<item value="money" quantity="[99,100]" probability="100.0"/>
```
After that change, the madaram healer always drops 99 money, but never 100.

*(mirrored on [SourceForge](https://sourceforge.net/p/arianne/bugs/5890/))*

**Comment by cosminhume** [comment]
I didn't check the implementation, but the first cause I suspect is that [`java.util.Random::nextInt(int)`](https://docs.oracle.com/javase/8/docs/api/java/util/Random.html#nextInt-int-) returns a _"value between 0 (inclusive) and the specified value (exclusive)"_. For example, `Random.nextInt(100)` would return minimum `0` and maximum `99`.

**Comment by AntumDeluge** [comment]
Fix committed to master: 2b069c9

Use [games.stendhal.common.Rand.randUniform](https://github.com/arianne/stendhal/blob/2b069c9d1e079735c0a0902ff49151fb8f5c55d4/src/games/stendhal/common/Rand.java#L78) instead of [games.stendhal.common.Rand.rand](https://github.com/arianne/stendhal/blob/2b069c9d1e079735c0a0902ff49151fb8f5c55d4/src/games/stendhal/common/Rand.java#L92).

**Comment by AntumDeluge** [comment]
Thank you for the response @cosminhume. Turns out that the project already had a better option available, just wasn't being used.
